디렉토리 구조 변경

chanha-park opened the pull request:
`[(main): [@login: [otp/], @home: [dropdown/, game/, social/]], auth/]` 대략 이런 구조로 변경했습니다.
(main)은 [routing group](https://nextjs.org/docs/app/building-your-application/routing/route-groups), @login과 @home은 [parallel route](https://nextjs.org/docs/app/building-your-application/routing/parallel-routes). @home은 기존 @main 디렉토리에 해당

변경 사유
1. 1차적으로 /main 안에 @main으로 이름이 겹치는 디렉토리 존재.
2. 첫 화면의 game start버튼으로 인해 불필요하게 경로 깊이가 한단계 깊어짐. (이건 나중에 어떻게든 추가를...)

따라서 중간중간 버튼이나 링크의 href값도 바꾼다고 바꿨으나.... 미처 못바꾼 부분이 있을 수 있습니다. 보시면 제보 부탁드립니다.

chanha-park commented:
덤으로 알려드리는 사항. 이런 디렉토리 구조 변경은 nextjs에서 잘 못해주는것 같네요 혹시 안되거나 404가 뜬다면 한번 껐다 켜보시길 
+ 현재 @main/layout.tsx가 적용이 안되는 이슈 있음

hdoo42 reviewed (approved):
너무 리뷰를 늦게 했네요..! 좋습니다

chanha-park commented:
이거 layout 수정 더 해야해서 draft로 바꿔놨었어요 ㅋㅋ 메인브랜치에 바뀐 파일도 많을거라... git rebase main해서 다시 올리든 해야하지 않을까 생각중이었어요

nimo-my reviewed (approved):
최고에요! 디렉토리 공부 좀 더 해놔야 겠어요.. 감사합니다!

chanha-park committed:
디렉토리 구조 변경

[(main): [@login: [otp/], @home: [dropdown/, game/, social/]], auth/]
대략 이런 구조로 변경.
(main)은 routing group, @login과 @home은 parallel route
@home은 기존 @main 디렉토리에 해당

chanha-park committed:
NavigationBar에서 async 키워드 제거

/(main)/@home/layout.tsx와 /(main)/@login/layout.tsx의 레이아웃을 상위
레벨의 layout으로 통합

chanha-park commented:
최근 main브랜치로 rebase해서 가져와서 고쳤습니다
